src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -5,24 +5,202 @@
  */
 package io.debezium.connector.cassandra;
 
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.config.CFMetaData;
+import org.apache.cassandra.config.Schema;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.schema.KeyspaceParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.SchemaChangeListener;
+import com.datastax.driver.core.TableMetadata;
+
+import io.debezium.connector.SourceInfoStructMaker;
+
 /**
- * The schema processor is responsible for periodically
- * refreshing the table schemas in Cassandra. Cassandra
- * CommitLog does not provide schema change as events,
- * so we pull the schema regularly for updates.
+ * The schema processor is reponsible for handling changes occuring in
+ * Cassandra via registered schema change listener into driver.
+ *
+ *
  */
 public class SchemaProcessor extends AbstractProcessor {
 
+    private static final Logger logger = LoggerFactory.getLogger(SchemaProcessor.class);
+
     private static final String NAME = "Schema Processor";
     private final SchemaHolder schemaHolder;
+    private final CassandraClient cassandraClient;
+    private final String kafkaTopicPrefix;
+    private final SourceInfoStructMaker sourceInfoStructMaker;
+    private final SchemaChangeListener schemaChangeListener;
 
     public SchemaProcessor(CassandraConnectorContext context) {
         super(NAME, context.getCassandraConnectorConfig().schemaPollInterval());
         schemaHolder = context.getSchemaHolder();
+        this.cassandraClient = context.getCassandraClient();
+        this.kafkaTopicPrefix = schemaHolder.kafkaTopicPrefix;
+        this.sourceInfoStructMaker = schemaHolder.sourceInfoStructMaker;
+
+        schemaChangeListener = new NoOpSchemaChangeListener() {
+            @Override
+            public void onKeyspaceAdded(final KeyspaceMetadata keyspace) {
```

Comment:
well that might be true in general but without having a keyspace added, the following update / creation of a table will not be possible because the underlying keyspace will not be there as we would just ignore its creation as you propose. The keyspace has to be there for the next tables you are going to possibly add afterwards.